pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/retention/RetentionManager.java

```diff
@@ -258,10 +257,11 @@ private void manageSegmentLineageCleanupForTable(TableConfig tableConfig) {
         if (SegmentLineageAccessHelper
             .writeSegmentLineage(_pinotHelixResourceManager.getPropertyStore(), segmentLineage, expectedVersion)) {
           // Delete segments based on the segment lineage
-          _pinotHelixResourceManager.deleteSegments(tableNameWithType, segmentsToDelete);
-          LOGGER.info("Finished cleaning up segment lineage for table: {}, deleted segments: {} in {}ms",
-              tableNameWithType, StringUtils.join(segmentsToDelete, ","),
-              (System.currentTimeMillis() - cleanupStartTime));
+          if (!segmentsToDelete.isEmpty()) {
+            _pinotHelixResourceManager.deleteSegments(tableNameWithType, segmentsToDelete);
+            LOGGER.info("Finished cleaning up segment lineage for table: {}, deleted segments: {} in {}ms",
```

Comment:
I'd suggest putting the duration before the `segmentToDelete`, as the list of segments to be deleted could be large.

Reply:
Changed. Can you double check?